core-bundle/src/Routing/Content/PageResolver.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\Content;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\PageModel;
+
+class PageResolver implements ContentUrlResolverInterface
+{
+    public function __construct(private readonly ContaoFramework $framework)
+    {
+    }
+
+    public function resolve(object $content): ContentUrlResult
+    {
+        if (!$content instanceof PageModel) {
+            return ContentUrlResult::abstain();
+        }
+
+        switch ($content->type) {
+            case 'redirect':
+                return ContentUrlResult::url($content->url);
+
+            case 'forward':
+                $pageAdapter = $this->framework->getAdapter(PageModel::class);
+
+                if ($content->jumpTo) {
+                    $forwardPage = $pageAdapter->findPublishedById($content->jumpTo);
```

Comment:
What if the target page no longer exists?

Reply:
`ContentUrlResult::redirect` will throw a `ForwardPageNotFound` exception if the forward target is `null`. This way not everyone has to handle the same case.

https://github.com/contao/contao/pull/6596/files#diff-984c96268523339467edac8695ebd6939603adfe8418c38e32f17e37f85152ceR86-R90